response_operations_ui/controllers/cir_controller.py

```diff
@@ -0,0 +1,79 @@
+import json
+import logging
+
+import requests
+from flask import current_app as app
+from google.auth.exceptions import GoogleAuthError
+from structlog import wrap_logger
+
+from response_operations_ui.common.credentials import fetch_and_apply_oidc_credentials
+from response_operations_ui.exceptions.error_codes import (
+    ErrorCode,
+    get_error_code_message,
+)
+from response_operations_ui.exceptions.exceptions import ExternalApiError
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+TARGET_SERVICE = "cir"
+
+
+def get_cir_service_status():
+    return _get_response_content(app.config["CIR_API_URL"] + "/status")
+
+
+def _get_response_content(request_url):
```

Comment:
This method acts as an exemplar to how we want to handle client requests to APIs. It deals explicitly with a number of well known negative scenarios. It's intended to be used (and extended as required) as a strategy for similar requests to this (and other) services.

Reply:
Makes sense.